src/paragraph.js

```diff
@@ -38,5 +49,174 @@ export default class Paragraph extends Plugin {
 		buildViewConverter().for( data.viewToModel )
 			.fromElement( 'p' )
 			.toElement( 'paragraph' );
+
+		// Autoparagraph text.
+		data.viewToModel.on( 'text', ( evt, data, consumable, conversionApi ) => {
+			autoparagraphText( doc, evt, data, consumable, conversionApi );
+		}, { priority: 'lowest' } );
+
+		// Post-fix potential subsequent paragraphs created by autoparagraphText().
+		data.viewToModel.on( 'element', mergeSubsequentParagraphs, { priority: 'lowest' } );
+		data.viewToModel.on( 'documentFragment', mergeSubsequentParagraphs, { priority: 'lowest' } );
+
+		// Convert paragraph-like elements to paragraphs if they weren't consumed.
+		// It's a 'low' priority in order to hook in before the default 'element' converter
+		// which would then convert children before handling this element.
+		data.viewToModel.on( 'element', ( evt, data, consumable, conversionApi ) => {
+			autoparagraphParagraphLikeElements( doc, evt, data, consumable, conversionApi );
+		}, { priority: 'low' } );
 	}
 }
+
+/**
+ * List of element names which should be treated by the autoparagraphing algorithms as
+ * paragraph-like. This means that e.g. the following content:
+ *
+ *		<h1>Foo</h1>
+ *		<table>
+ *			<tr>
+ *				<td>X</td>
+ *				<td>
+ *					<ul>
+ *						<li>Y</li>
+ *						<li>Z</li>
+ *					</ul>
+ *				</td>
+ *			</tr>
+ *		</table>
+ *
+ * Contains five paragraph-like elements – `<h1>` and two `<td>` and two `<li>`.
+ * Hence, if none of the features is going to convert  those elements the above content will be automatically handled
+ * by the paragraph feature and converted to:
+ *
+ *		<p>Foo</p>
+ *		<p>X</p>
+ *		<p>Y</p>
+ *		<p>Z</p>
+ *
+ * Note: The `<td>` containing two `<li>` elements was ignored – the inner-most paragraph-like elements
+ * have priority upon conversion.
+ *
+ * @member {Set.<String>} module:paragraph/paragraph~Paragraph.paragraphLikeElements
+ */
+Paragraph.paragraphLikeElements = new Set( [
+	'p',
+	'h1',
+	'h2',
+	'h3',
+	'h4',
+	'h5',
+	'h6',
+	'td',
+	'li',
+	'div',
+	'dt',
+	'dd',
+	'blockquote'
+] );
```

Comment:
By `<td>` which is in this table too (and is tested in automated and manual tests).